cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityReader.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import io.cdap.cdap.api.app.ApplicationSpecification;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * CapabilityReader interface with methods based on current capability status
+ */
+public interface CapabilityReader {
+
+  /**
+   * Returns the status of capability.
+   *
+   * @param capability
+   * @return
+   * @throws IOException
+   */
+  CapabilityStatus getStatus(String capability) throws IOException;
+
+  /**
+   * Return boolean indicating whether the capability is enabled
+   *
+   * @param capability
+   * @return
+   * @throws IOException
+   */
+  boolean isEnabled(String capability) throws IOException;
+
+  /**
+   * Throws {@link CapabilityNotAvailableException} if all capabilities are not enabled
+   * @param capabilities
+   * @return
+   * @throws IOException
+   */
+  void checkAllEnabled(Collection<String> capabilities) throws IOException, CapabilityNotAvailableException;
+
+  /**
+   * Throws {@link CapabilityNotAvailableException} if all capabilities are not enabled
+   * @param appSpec
+   * @return
+   * @throws IOException
+   */
+  void checkAllEnabled(ApplicationSpecification appSpec) throws IOException, CapabilityNotAvailableException;
+
+  /**
+   * Get the configuration for this capability if present
+   *
+   * @param capability
+   * @return
+   * @throws IOException
+   */
+  @Nullable
+  CapabilityConfig getConfig(String capability) throws IOException;
```

Comment:
Avoid expose method that backed by DB and have the caller looping and calling it in a tight loop. It's better to take a collection and returns a map/collection.

Reply:
Done